pip/pip-321.md

```diff
@@ -0,0 +1,251 @@
+
+Add two new namespace policies `allowed_clusters`, and `topic_policy_synchronized_clusters`.
+Replace the function of  `replication_clusters` with `replication_clusters`, `allowed_clusters`, and `topic_policy_synchronized_clusters`.
+
+# Detailed Design
+
+## Public-facing Changes
+
+### Public API
+Add a new endpoint `setNamespaceAllowedClusters` to the REST API.
+```
+Type: POST
+Path: /namespaces/{tenant}/{namespace}/allowedcluster
+Query parameters: tenant, namespace
+HTTP body parameters: List<String> clusterIds
+Response code:
+  * 400 Bad request, Allowed-clusters should contain all replication-clusters
+  * 403 Forbidden, Don't have admin permission
+  * 404 Not Found, Tenant or cluster or namespace doesn't exist.
+  * 409 Conflict, Peer-cluster can't be part of allowed-cluster.
+  * 412 Precondition Failed, Namespace is not global or invalid cluster ids.
+Explanation for 409 Conflict: This follows the behavior of namespace replication clusters. In the PIP-8, quoted in background knowledge, it introduce the concept of peer-cluster and peer-cluster can't be part of replication-cluster. For the allowed-clusters, users could enable replication at the topic level, so the peer-cluster can't be part of allowed-clusters too.
+```
+
+Add a new endpoint `getNamespaceAllowedClusters` to the REST API.
+```
+type: GET
+Path: /namespaces/{tenant}/{namespace}/allowedcluster
+Query parameters: tenant, namespace
+Response code:
+  * 403 Forbidden, Don't have admin permission
+  * 404 Not Found, Tenant or cluster or namespace doesn't exist.
+  * 412 Precondition Failed, Namespace is not global.
+```
+
+Add a new endpoint `setNamespaceTopicPolicySynchronizedClusters` to the REST API.
+```
+Type: POST
+Path: /namespaces/{tenant}/{namespace}/topicpolicycluster
+Query parameters: tenant, namespace
+HTTP body parameters: List<String> clusterIds
+Response code:
+  * 400 Bad request, Allowed-clusters should contain all topic-policy-synchronized-clusters
+  * 403 Forbidden, Don't have admin permission
+  * 404 Not Found, Tenant or cluster or namespace doesn't exist.
+  * 412 Precondition Failed, Namespace is not global or invalid cluster ids.
+```
+
+Add a new endpoint `getNamespaceTopicPolicySynchronizedClusters` to the REST API.
+```
+Type: GET
+Path: /namespaces/{tenant}/{namespace}/topicpolicycluster
+Query parameters: tenant, namespace
+Response code:
+  * 403 Forbidden, Don't have admin permission
+  * 404 Not Found, Tenant or cluster or namespace doesn't exist.
+  * 412 Precondition Failed, Namespace is not global.
+```
+
+### Binary protocol
+
+### Configuration
+
+### CLI
+
+### Metrics
+None.
+
+# Monitoring
+None.
+# Security Considerations
+If the broker enables authentication, then this configuration can only be set by the client who was authenticated.
+
+# Backward & Forward Compatibility
+
+## Revert
+This can not be reverted if they enable geo-replication at the topic level. Otherwise, the replicator of these topics will be closed.
```

Comment:
I think the revert here should be for the rollback (from the new configuration to the old configuration) or rollback to an old version.